src/app/dev/DevToys.Tools/Tools/Text/ListCompare/ListCompareGuiTool.cs

```diff
@@ -0,0 +1,215 @@
+                                            Item(ListCompare.AInterB, ListComparisonMode.AInterB),
+                                            Item(ListCompare.AUnionB, ListComparisonMode.AUnionB),
+                                            Item(ListCompare.AOnly, ListComparisonMode.AOnly),
+                                            Item(ListCompare.BOnly, ListComparisonMode.BOnly)
+                                )
+                            )
+                        ),
+
+
+                    Cell(
+                        GridRows.Text,
+                        GridColumns.Stretch,
+
+                        SplitGrid()
+                            .Horizontal()
+                            .TopPaneLength(new UIGridLength(1, UIGridUnitType.Fraction))
+                            .BottomPaneLength(new UIGridLength(2, UIGridUnitType.Fraction))
+                            .WithTopPaneChild(
+                                SplitGrid()
+                                    .Vertical()
+                                    .WithLeftPaneChild(
+                                        MultilineTextInput()
+                                            .Title(ListCompare.ListA)
+                                            .OnTextChanged(OnListAChanged))
+                                    .WithRightPaneChild(
+                                        MultilineTextInput()
+                                            .Title(ListCompare.ListB)
+                                            .OnTextChanged(OnListBChanged)))
+                            .WithBottomPaneChild(
+                                _diffListResult
+                                    .ReadOnly()
+                                    .Extendable())
+                            )));
+
+    public void OnDataReceived(string dataTypeName, object? parsedData)
+    {
+    }
+
+    private ValueTask OnCaseSensitiveChanged(bool caseSensitive)
+    {
+        StartCompare();
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnListAChanged(string listA)
+    {
+        _listA = listA;
+        StartCompare();
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnListBChanged(string listB)
+    {
+        _listB = listB;
+        StartCompare();
+        return ValueTask.CompletedTask;
+    }
+
+    private void OnComparisonModeChanged(ListComparisonMode col)
+    {
+        StartCompare();
+    }
+
+    private void StartCompare()
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _cancellationTokenSource = new CancellationTokenSource();
+
+        WorkTask = CompareAsync(_listA, _listB, _settingsProvider.GetSetting(caseSensitive), _settingsProvider.GetSetting(comparisonMode), _cancellationTokenSource.Token);
+        string selectedComparisonModeTitle = _settingsProvider.GetSetting(comparisonMode) switch
```

Comment:
This Ui update can be moved to `CompareAsync`, near where `_diffListResult.Text` is called.